streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -267,12 +267,19 @@ void initializeStateStores() {
      * </pre>
      */
     public void resume() {
-        log.debug("Resuming");
-        if (state() == State.SUSPENDED) {
-            initializeMetadata();
-            transitionTo(State.RUNNING);
-        } else {
-            throw new IllegalStateException("Expected only to resume from SUSPENDED state, but was " + state());
+        switch (state()) {
+            case RUNNING:
```

Comment:
We may call assignor#onAssignment multiple times without `onPartitionsRevoked`, therefore not all existing active tasks are in SUSPENDED when we are `handleAssignment`ing. For those that are still in RUNNING / RESTORING we just do nothing.